### ISSUE TITLE
Use standard soname for library products

### ISSUE THREAD

**Issue by sloede** [DLR-AMR/t8code#483]
Right now, when building t8code, the library products `libt8.so`, `libsc.so`, and `libp4est.so` have the full version and some hash encoded in the filename. For Linux, the soname for a library should be something like `libLIBRARYNAME.so.X.Y.Z`, where `X.Y.Z` are *not* the library's release number but rather the *API version*. That is, the version should only change if there are backwards incompatible changes to the library API (see e.g., https://tldp.org/HOWTO/Program-Library-HOWTO/shared-libraries.html).

Right now, t8code produces a soname with its version and some has encoded in it, e.g., `libt8-1.1.0.207-d6a74.so`, which is then linked symbolically to the generic `libt8.so` (similar for libp4est and libsc). This makes it somewhat inconvenient to use, as you can only link to a single library version at a time (see also the remark by @giordano at https://github.com/JuliaPackaging/Yggdrasil/pull/6389#issuecomment-1475283561). When building p4est itself, it actually does what is expected: it creates a file named `libp4est.so.0.0.0`, which is linked to `libp4est.so.0`, which is linked to `libp4est.so` (similarly for libsc).

Long story short, I think it would be beneficial to change the build system of t8code to reflect this Unix/Linux convention of how libraries are named. At the moment, though, I have not encountered any problems myself yet, so for me this is not an urgent issue (yet) 🙂

cc @jmark

**Comment by cburstedde** [comment]
> Long story short, I think it would be beneficial to change the build system of t8code to reflect this Unix/Linux convention of how libraries are named. At the moment, though, I have not encountered any problems myself yet, so for me this is not an urgent issue (yet) 🙂

It is sufficient to add -version-info to LDFLAGS of the library in Makefile.am.

Please see how I just did this with the oldcompat branches of p4est and libsc.

The libtool manual has a whole section on versioning.

**Comment by sloede** [comment]
Interesting. However, I do not understand why the current versions of t8code and p4est behave differently, if this flag has not been used before on the p4est side. That is, why does p4est 2.8 install "correct" library names, while t8code encodes the full version has. Do you have an idea how this could happen?

**Comment by cburstedde** [comment]
I've added API version 2:0:0 to the latest p4est and libsc, which previously defaulted to 0.
I'm adding the same API version to t8code in #499 to start fresh with the numbering.

In t8code, in addition to -version-info there is -release $(VERSION), which may be needed or not (?).

Whan's still missing imho is to set the SO version with CMake for all three libraries.

**Comment by sloede** [comment]
OK, great. Let's wait for #499 being merged and then revisit the question of proper SONAMEs for t8code afterwards

**Comment by ole-alb** [comment]
PR that should've closed the Issue became obsolete, because the changes have been merged otherwise. Therefore this Issue also becomes obsolete.
